examples/fluids/qfunctions/newtonian_state.h

```diff
@@ -166,6 +166,24 @@ CEED_QFUNCTION_HELPER State StateFromY_fwd(NewtonianIdealGasContext gas,
   return ds;
 }
 
+// Generic State converters based on gas->use_primitive
+CEED_QFUNCTION_HELPER State StateFromQi(NewtonianIdealGasContext gas,
+                                        const CeedScalar Qi[5], const CeedScalar x[3]) {
+  if (gas->use_primitive) return StateFromY(gas, Qi, x);
+  else                    return StateFromU(gas, Qi, x);
+}
```

Comment:
I'm okay with this, though I can't say I love having the two different conventions.

Reply:
I'm not married to having the different conventions. I honestly don't like having different ones, but I'm being a bit lazy. I can change them to be all pointer based (I'd probably make a typedef for `StateFromQi*` function pointers to ease the verbosity in the helper function definitions).

Reply:
I've just pushed up changes to replace `StateFromQi` with function pointers. Given the large change, I'd appreciate a quick glance to make sure I haven't done anything too nefarious.